### PR TITLE
fix: URL query not passed on when there's only one EventType is active

### LINKED ISSUE #12758: URL parameters not passed on when only one Event Type is active
Found a bug? Please fill out the sections below. 👍

### Issue Summary

A summary of the issue. This needs to be a clear detailed-rich summary.

### Steps to Reproduce

1. Leave only one event type active for a user
2. Navigate to the user's main calendar url and include some parameters e.g., cal.com/username?email=user.com
3. Due to the auto redirect, the URL parameters are not passed to the booking page

This piece of code is relevant:

apps/web/pages/[user].tsx (line 393, as of version 3.5.4)

`  // if profile only has one public event-type, redirect to it
  if (eventTypes.length === 1 && context.query.redirect !== "false") {
  
    return {
    
      redirect: {
        permanent: false,
        destination: `/${user.username}/${eventTypes[0].slug}`,
      },
    };
  }`

### PR DESCRIPTION
**Pull request by wtLau**
## What does this PR do?

Fixes #12758

https://www.loom.com/share/26aa36c87fed4127ba57a2dfc25990aa?sid=ad43cea2-3e7a-4943-b0ed-03b20287137c
## Requirement/Documentation

Steps to Reproduce
- Leave only one event type active for a user
- Navigate to the user's main calendar url and include some parameters e.g., cal.com/username?email=user.com
- Due to the auto redirect, the URL parameters are not passed to the booking page

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by wtLau** [comment]
Please let me know if tests are needed for merging! Happy to add it.

File: apps/web/pages/[user].tsx [hunk 1]
```diff
@@ -391,10 +392,16 @@ export const getServerSideProps: GetServerSideProps<UserPageProps> = async (cont
...
+      // So it doesn't display in the Link
+      user: _user,
+      ...query
+    } = context.query;
+
     return {
       redirect: {
         permanent: false,
-        destination: `/${user.username}/${eventTypes[0].slug}`,
+        destination: `/${user.username}/${eventTypes[0].slug}?${stringify(query)}`,
```
Comment (Udit-takkar) [inline apps/web/pages/[user].tsx:404]: I don't think we want this. Prefilling is already supported for event types url /[user]/[type].

**Comment by emrysal** [comment]
As the URL may actually change per user when event types change, this is not a permalink, instead it's a dynamic link that will change over time and this is the reason I'd prefer not to allow this option. If you want URL queries - you should directly link to the event type.

**Comment by wtLau** [comment]
Thank you for reviewing!

**Pull request closed without merging**